### DISCUSSION TITLE
[Spring] 다대다 연결관계에서 연결엔티티 사용방법

### DISCUSSION THREAD

**Discussion by DoohyunHwang97** [wafflestudio/seminar-2023#161] [Q&A] [answered]
<img width="440" alt="image" src="https://github.com/wafflestudio/seminar-2023/assets/76721027/9864c829-7a47-4b92-8b27-1b2090cd1e1c">
<img width="479" alt="image" src="https://github.com/wafflestudio/seminar-2023/assets/76721027/5164648c-9b12-474b-a994-af9600cb8499">
<img width="460" alt="image" src="https://github.com/wafflestudio/seminar-2023/assets/76721027/8aaba4a5-e87a-43fa-8cfc-956aa4935743">

<img width="802" alt="image" src="https://github.com/wafflestudio/seminar-2023/assets/76721027/8127248c-adf9-47d9-82f6-fd73126ddc9d">

ER 다이어그램에서 제시된 것처럼 song과 artist 사이 관계를 연결엔티티를 이용해서 구현하려합니다.
하지만 songEntity를 통해서 연관된 List<Artist>를 얻고 싶은 상황에서 
<img width="1496" alt="image" src="https://github.com/wafflestudio/seminar-2023/assets/76721027/680fa6e2-87dd-4944-ae3e-4cdb61ab87b0">
다음과 같은 상황을 맞이했습니다. 궁극적으로 List<Artist>를 얻기 위한 방법을 고민해봤지만 이를 ORM 문법적으로 해결할 수 있는 방법이 있는 것인지 아니면 JPQL로 해결할 수 있는 것인지조차 감이 잡히지 않아서 질문 드립니다.

**Comment by DoohyunHwang97** [comment]
<img width="865" alt="image" src="https://github.com/wafflestudio/seminar-2023/assets/76721027/b7140f15-b3c0-447e-a83e-2c6fae2457b6">
컬렉션 map을 이용해서 이러한 방식으로 구현해봤는데 올바른 방법인지도 질문 드립니다!

**Reply by PFCJeong** [reply]
넵. 제가 의도한 방법입니다 👍

**Reply by DoohyunHwang97** [reply]
감사합니다 :)

**Comment by rayark1** [comment, marked as answer]
현재 중계테이블을 사용하셔서 songEntity.song_artists를 사용하시면 songEntity에 정의된대로, 해당 송 id가 포함되어있는 song_artists 중계 테이블 songArtistEntity가 호출된 것 같습니다. 따라서, 정의 하신 SongArtistEntity를 이용하여 artist를 호출하시면 될 것 같습니다! 자료형이 list일 경우 map을 이용해서 처리하시면 됩니다.

**Reply by DoohyunHwang97** [reply]
감사합니다 :)
